### PR TITLE
Add support for U55N

### PR DESCRIPTION
**Pull request by HongshiTan**
* refactor ping_fpga example for a better config of ip address.
* add dump of arp table in ping_fpga example

### CODE REVIEW

File: Ethernet/platform.tcl [hunk 1]
```diff
@@ -5,16 +5,20 @@
...
-} elseif {[string first "u55" ${board}] != -1} {
+} elseif {[string first "u55c" ${board}] != -1} {
     set projPart "xcu55c-fsvh2892-2L-e"
 } elseif {[string first "u200" ${board}] != -1} {
     set projPart "xcu200-fsgd2104-2-e"
 } elseif {[string first "u250" ${board}] != -1} {
     set projPart "xcu250-figd2104-2L-e"
 } elseif {[string first "u280" ${board}] != -1} {
     set projPart "xcu280-fsvh2892-2L-e"
+} elseif {[string first "vck5000" ${board}] != -1} {
```
Comment (mariodruiz) [inline Ethernet/platform.tcl:20]: There's no current plan to support VCK5000, please remove this

File: xrt_host_api/examples/ping_fpga/ping_fpga.cpp [hunk 2]
```diff
@@ -19,31 +19,88 @@ namespace fs = std::filesystem;
...
-const char *hostname3 = "hostname3";
 
-const std::map<std::string, const std::vector<std::string>> ip_addresses = {
-    {hostname1, {"10.1.212.101", "10.1.212.102"}},
-    {hostname2, {"10.1.212.103", "10.1.212.104"}},
-    {hostname3, {"10.1.212.105", "10.1.212.106"}}};
+ip_config_table_t * get_ip_config(std::string hostname, uint32_t board_id)
+{
+    ip_config_table_t * default_ip_config = &ip_lut[0]; // will set default to the first matched hostname.
+    for (uint32_t  i = 0; i < ARRAY_SIZE(ip_lut); i++)
```
Comment (mariodruiz) [inline xrt_host_api/examples/ping_fpga/ping_fpga.cpp:58]: To keep consistency put the curly brace `{` at the same level of the for

File: xrt_host_api/examples/ping_fpga/ping_fpga.cpp [hunk 3]
```diff
@@ -19,31 +19,88 @@ namespace fs = std::filesystem;
...
+    {
+        if(std::string(ip_lut[i].hostname) ==  hostname)
```
Comment (mariodruiz) [inline xrt_host_api/examples/ping_fpga/ping_fpga.cpp:60]: ditto

File: xrt_host_api/examples/ping_fpga/ping_fpga.cpp [hunk 4]
```diff
@@ -19,31 +19,88 @@ namespace fs = std::filesystem;
...
+        {
+            if(ip_lut[i].board_id == board_id)
+            {
```
Comment (mariodruiz) [inline xrt_host_api/examples/ping_fpga/ping_fpga.cpp:63]: ditto

File: xrt_host_api/examples/ping_fpga/ping_fpga.cpp [hunk 5]
```diff
@@ -156,25 +208,35 @@ int main(int argc, char *argv[]) {
...
-    ++ip_index;
+    std::string ip = get_ip_config(hostname, device_id)->ip_address[ip_index];
 
     std::cout << "setting up IP " << ip << " to interface " << cus.first
               << std::endl;
     networklayer.update_ip_address(ip);
     std::this_thread::sleep_for(std::chrono::seconds(1));
 
+    for(int j = 0;  j < 5 ; j++)
+    {
```
Comment (mariodruiz) [inline xrt_host_api/examples/ping_fpga/ping_fpga.cpp:219]: ditto

File: xrt_host_api/examples/ping_fpga/ping_fpga.cpp [hunk 6]
```diff
@@ -156,25 +208,35 @@ int main(int argc, char *argv[]) {
...
+        networklayer.arp_discovery();
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+
+    }
+
+    auto table = networklayer.read_arp_table(255);
+    std::cout <<"Dump arp table" << std::endl;
+    for (const auto& [id, value] : table)
+    {
```
Comment (mariodruiz) [inline xrt_host_api/examples/ping_fpga/ping_fpga.cpp:228]: ditto

File: xrt_host_api/examples/ping_fpga/ping_fpga.cpp [hunk 7]
```diff
@@ -156,25 +208,35 @@ int main(int argc, char *argv[]) {
...
 
+    for(int j = 0;  j < 5 ; j++)
+    {
+        networklayer.arp_discovery();
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+
+    }
+
+    auto table = networklayer.read_arp_table(255);
+    std::cout <<"Dump arp table" << std::endl;
```
Comment (mariodruiz) [inline xrt_host_api/examples/ping_fpga/ping_fpga.cpp:226]: What is the value of dumping the ARP table by default? This could be a paramenter

**Review by mariodruiz — CHANGES_REQUESTED** [review]
Hi @HongshiTan,

Thank you for your contribution, please address my comments

File: Ethernet/platform.tcl (hunk 1 above)
Comment (HongshiTan) [reply Ethernet/platform.tcl]: OK. I'm doing the test on VCK5000 now, will push it when finished.

**Comment by HongshiTan** [comment]
Have resolved all your comments

**Review by mariodruiz — APPROVED** [review]

**Pull request merged**